### PR TITLE
Add some UI tests

### PR DESCRIPTION
**Pull request by johnstill**
This branch was going to include more tests but I'd rather merge now and then merge other tests later than keep waiting.

### CODE REVIEW

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        ],
+        providers: [
+          ReviewStateService,
+          {provide: CohortAnnotationDefinitionService, useValue: new ApiSpy()},
+          {provide: ActivatedRoute, useClass: StubRoute},
+        ],
+      }).compileComponents().then((resp) => {
+        fixture = TestBed.createComponent(SetAnnotationItemComponent);
+
+        // tick();
```
Comment (calbach) [inline ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: rm?

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        component = fixture.componentInstance;
+
+        // Default Inputs for tests
+        component.definition = <CohortAnnotationDefinition>{
```
Comment (calbach) [inline ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: nit: is the cast needed here? I would have thought not (but not opposed if it's just for being explicit, mostly curious)

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+          cohortAnnotationDefinitionId: 1,
+          cohortId: 1,
+          columnName: 'Test Defn',
+          annotationType: AnnotationType.STRING,
+        };
+        updateAndTick(fixture);
+      });
+    // tick();
```
Comment (calbach) [inline ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: rm?

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+  it('Should render', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('Should set the form focus when editing', fakeAsync(() => {
+    component.editing = true;
+    updateAndTick(fixture);
+    component.setFocus();
+    // component.edit() calls setTimeout so there's still a timer on the stack
```
Comment (calbach) [inline ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: Is the comment outdated? I don't follow where `component.edit()` is being called

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  it('Should set the form value to the existing definition value', fakeAsync(() => {
+    // Monkey patch this method b/c it misbehaves in the test environment
+    // unless treated in isolation
+    component.setFocus = () => {};
+    updateAndTick(fixture);
+    component.edit();
+    expect(component.definition.columnName).toEqual(component.name.value);
+  }));
+
+  it('Should refuse to save an empty name value', fakeAsync(() => {
```
Comment (calbach) [inline ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: Should there be a test case where the updateCohortAnnotationDefinition actually gets called?

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.ts [hunk 6]
```diff
@@ -43,9 +43,13 @@ export class SetAnnotationItemComponent {
   edit(): void {
     this.editing = true;
     this.name.setValue(this.definition.columnName);
+    this.setFocus();
+  }
+
+  setFocus() {
```
Comment (blrubenstein) [inline ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.ts]: Can we/should we put setFocus in test-helpers? Do we think it is going to see a lot of use in many different components?

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+          columnName: 'Test Defn',
+          annotationType: AnnotationType.STRING,
+        };
+        updateAndTick(fixture);
+      });
+    // tick();
+  }));
+
+  it('Should render', () => {
+    expect(component).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (johnstill) [reply ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: Its just kind of a sanity check I like to have in each test file - every other test should presuppose that the component exists.  So if this fails I know immediately something is wrong with the testing harness (or fundamentally broken with the component).

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts (hunk 4 above)
Comment (johnstill) [reply ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: The comment was indeed outdated; the contents of `setFocus` used to be in `edit`

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.ts (hunk 6 above)
Comment (johnstill) [reply ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.ts]: It probably won't get a lot of use actually, its a pretty specific condition that this solves.  We'd have to do some work to remove it from the component b/c it relies on closures, but if we find that its being used a lot we could do that.

Why would it go into test-helpers though?  Its a component method.

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts (hunk 3 above)
Comment (johnstill) [reply ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: Yes.

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts (hunk 1 above)
Comment (johnstill) [reply ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: Yes.

File: ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts (hunk 5 above)
Comment (johnstill) [reply ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts]: Probably, haha.  I'll push that up next.

**Pull request merged**